src/provider/proxyprovider.cpp

```diff
@@ -70,3 +70,45 @@ TransactionStatus ProxyProvider::getTransactionStatus(std::string const &txHash)
 
     return TransactionStatus(txStatus);
 }
+
+std::string ProxyProvider::getESDTTokenBalance(Address const &address, std::string const &token) const
+{
+    wrapper::http::Client client(m_url);
+    wrapper::http::Result const result = client.get("/address/" + address.getBech32Address() + "/esdt/" + token);
+
+    auto data = internal::getPayLoad(result);
+
+    utility::requireAttribute(data, "tokenData");
+    utility::requireAttribute(data["tokenData"], "balance");
+
+    std::string balance = data["tokenData"]["balance"];
+
+    return balance;
+}
+
+std::map<std::string, std::string> ProxyProvider::getAllESDTTokenBalances(Address const &address) const
+{
+    wrapper::http::Client client(m_url);
+    wrapper::http::Result const result = client.get("/address/" + address.getBech32Address() + "/esdt");
+
+    auto data = internal::getPayLoad(result);
+
+    utility::requireAttribute(data, "esdts");
+
+    auto esdts = data["esdts"];
+
+    std::map<std::string, std::string> ret;
+    std::string esdt;
+    std::string balance;
+
+    for (const auto& it : esdts.items())
+    {
+        utility::requireAttribute(it.value(), "balance");
```

Comment:
Not sure, can we have some functionality such as:

```
utility::getAttribute(someData, attributeName, fallbackValueIfAttributeIsMissingOrNull)
```

Not in this PR, but in a future one, if it proves useful (e.g. in this case?)

Reply:
It would be useful as a fallback, but in this case I don't think we want that.

Suppose that somehow we get a ProxyResponse that doesn't contain a field, but we fill it with default value (e.g. default balance = "0"). For those who use the response, since we don't throw any exception, they will use that balance thinking it is correct.